server/src/main/java/keywhiz/service/permissions/PermissionCheckProvider.java

```diff
@@ -0,0 +1,21 @@
+package keywhiz.service.permissions;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+public class PermissionCheckProvider implements Provider<PermissionCheck> {
+  private final MetricRegistry metricRegistry;
+
+  @Inject
+  public PermissionCheckProvider(MetricRegistry metricRegistry) {
+    this.metricRegistry = metricRegistry;
+  }
+
+  @Override
+  public PermissionCheck get() {
+    PermissionCheck alwaysFail = new AlwaysFailPermissionCheck();
+    PermissionCheck alwaysAllow = new AlwaysAllowPermissionCheck(this.metricRegistry, alwaysFail);
+    return alwaysAllow;
+  }
+}
```

Comment:
Ok, I had to read around a bit to see how this is used. Binding to a `Provider<T>` and having `@ Provides T someMethod()` on a module are totally equivalent as far as creating instances of T, however, I think since our logic is simple (so far) it might be more clear to move this to an `@Provides` method on the module so that the logic/binding is all in one place.